filelog.go

```diff
@@ -128,12 +128,21 @@ func NewFileLogWriter(fname string, rotate bool) *FileLogWriter {
 			}
 		}()
 
+		// Failure counters
+		var rotationFailures uint64
+		var writeFailures uint64
+
 		for {
 			select {
 			case <-w.rot:
 				if err := w.handleRotate(time.Now()); err != nil {
 					fmt.Fprintf(os.Stderr, "FileLogWriter(%q): %s\n", w.filename, err)
-					return
+					rotationFailures += 1
+				} else {
+					if rotationFailures != 0 {
+						_, logErr := fmt.Fprintf(os.Stderr, "FileLogWriter(%q): Ignored %d previous rotation attempts due to rotation failure\n", w.filename, rotationFailures)
+						if logErr == nil { rotationFailures = 0 }
```

Comment:
Hmm, I guess we do need a reset of `rotationFailures`. But... I think I'm having a little trouble understand the flow.

Let me ask a diff question.  Do we need to print the "Ignored X failures to rotate" ever time we go from failing to succeeding (which I *think* is what's happening)?  Should this maybe be `if WeSuccededButTheLastAttemptFailed && WeWereUnableToPrintThatTheLastAttemptFailed { printStuff()`?